runner/status.go

```diff
@@ -26,12 +26,12 @@ const (
 	ABORTED
 	// Operation timed out and was killed.
 	TIMEDOUT
-	// Invalid or error'd request. Original runner state not affected. Retry may work after mutation.
+	// Request rejected for reasons unrelated to the user.
```

Comment:
Great comment. I wasn't particularly satisfied with my description. I would say that we can break things down into two categories - transient failures due to resource issues, and failures due to inconsistent state / internal logic.

Do the current errors match those categories in your opinion? Are those useful error categories or do you think we can add more state or just boil it all down to FAILED?